### PR TITLE
USAGOV-1676 handle the additional docker image digest for the www app

### PR DESCRIPTION
**Pull request by arpage**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1676

## Description

Add better error handling, as well as incorporate the www app docker image digest into the annotated git tag.

## Type of Changes

- [ ] New Feature
- [ ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [x] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [x] WWW
  - [ ] Tools
- [ ] Other

### Requires New Config
- [ ] Yes
- [ ] No

### Requires New Content
- [ ] Yes
- [ ] No

### Validation Steps
- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

**Comment by arpage** [comment]
@akf - this has been tested by running the scripts to add/check the image digests from last Tuesday's deployment to a tag on the prod branch.

Create tag:
```
❯ bin/deploy/git-annotated-tag-creator.sh prod 9668 @sha256:b0c62c221bdcae9a6964ca5c0c70780b018dc86405be3d0ecb74fd99d8153fbe @sha256:0517924b52f4da61c174b0d918d31130a3e05138ac73f82181b6589f49bd48aa @sha256:14bd23eda85ee79de8b771847d4485173226d7a038943f56eeb0b4e056c16b29
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 343 bytes | 343.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0)
To github.com:usagov/usagov-2021.git
 * [new tag]             usagov-cci-build-9668-prod -> usagov-cci-build-9668-prod

```

Retrieve tag:
```
❯ bin/deploy/git-annotation-parser.sh prod

To deploy the waf, please execute the following command:

   ROUTE_SERVICE_APP_NAME=waf \
     ROUTE_SERVICE_NAME=waf-route-prod-usagov \
     PROTECTED_APP_NAME=cms \
        bin/cloudgov/deploy-waf 9668 @sha256:0517924b52f4da61c174b0d918d31130a3e05138ac73f82181b6589f49bd48aa


To deploy the cms, please execute the following command:
   bin/cloudgov/deploy-cms 9668 @sha256:b0c62c221bdcae9a6964ca5c0c70780b018dc86405be3d0ecb74fd99d8153fbe


To deploy the static site, please execute the following command:
   bin/cloudgov/deploy-www 9668 @sha256:14bd23eda85ee79de8b771847d4485173226d7a038943f56eeb0b4e056c16b29

```

File: bin/deploy/git-annotation-parser.sh [hunk 1]
```diff
@@ -116,31 +116,65 @@ for at in $ANNOTATED_TAGS; do
...
+    echo
+    echo "To deploy the cms, please execute the following command:"
+    echo "   bin/cloudgov/deploy-cms $CCI_BUILD $CMS_DIGEST"
+    echo
+    echo
+    echo "To deploy the static site, please execute the following command:"
+    echo "   bin/cloudgov/deploy-www $CCI_BUILD $WWW_DIGEST"
+    echo
+  else
+     echo "Not all image digests were found in the tag: $ANNOTATED_TAG"
```
Comment (akf) [inline bin/deploy/git-annotation-parser.sh]: Minor nit: we can also get here if we don't have CCI_BUILD defined. Probably we should add an "echo" for that as well.

**Review by akf — COMMENTED** [review]
This looks good, although I did not actually test it. Do you want to address my one nit-pick (either by making an update, or telling me I'm wrong)?

File: bin/deploy/git-annotation-parser.sh (hunk 1 above)
Comment (arpage) [reply bin/deploy/git-annotation-parser.sh]: I have updated the script to properly handle/announce the case where CCI_BUILD is missing from the git tag annotation.

**Review by akf — APPROVED** [review]
Looks good! Worked on my Mac: 
```
% bin/deploy/git-annotation-parser.sh prod

To deploy the waf, please execute the following command:

   ROUTE_SERVICE_APP_NAME=waf \
       ROUTE_SERVICE_NAME=waf-route-prod-usagov \
       PROTECTED_APP_NAME=cms \
          bin/cloudgov/deploy-waf 9668 @sha256:0517924b52f4da61c174b0d918d31130a3e05138ac73f82181b6589f49bd48aa


To deploy the cms, please execute the following command:
   bin/cloudgov/deploy-cms 9668 @sha256:b0c62c221bdcae9a6964ca5c0c70780b018dc86405be3d0ecb74fd99d8153fbe


To deploy the static site, please execute the following command:
   bin/cloudgov/deploy-www 9668 @sha256:14bd23eda85ee79de8b771847d4485173226d7a038943f56eeb0b4e056c16b29
```

**Pull request merged**